azurerm/common_hdinsight.go

```diff
@@ -35,28 +35,34 @@ func hdinsightClusterUpdate(clusterKind string, readFunc schema.ReadFunc) schema
 				Tags: tags.Expand(t),
 			}
 			if _, err := client.Update(ctx, resourceGroup, name, params); err != nil {
-				return fmt.Errorf("Error updating Tags for HDInsight %q Cluster %q (Resource Group %q): %+v", clusterKind, name, resourceGroup, err)
+				return fmt.Errorf("error updating Tags for HDInsight %q Cluster %q (Resource Group %q): %+v", clusterKind, name, resourceGroup, err)
 			}
 		}
 
 		if d.HasChange("roles.0.worker_node") {
-			log.Printf("[DEBUG] Resizing the HDInsight %q Cluster", clusterKind)
-			rolesRaw := d.Get("roles").([]interface{})
-			roles := rolesRaw[0].(map[string]interface{})
-			workerNodes := roles["worker_node"].([]interface{})
-			workerNode := workerNodes[0].(map[string]interface{})
-			targetInstanceCount := workerNode["target_instance_count"].(int)
-			params := hdinsight.ClusterResizeParameters{
-				TargetInstanceCount: utils.Int32(int32(targetInstanceCount)),
-			}
+			oldWorkerNodeCount, newWorkerNodeCount := d.GetChange("roles.0.worker_node.0.target_instance_count")
+			oldWorkerNodeInt := oldWorkerNodeCount.(int)
+			newWorkerNodeInt := newWorkerNodeCount.(int)
+
+			if oldWorkerNodeInt != newWorkerNodeInt {
```

Comment:
Do you mind talking through this change. I believe that it's appropriate but a little context would be helpful in fully understanding why we're implementing this change

Reply:
We basically want to resize the cluster only when the target_instance_count has changed.

Because I added additional items to the worker_node schema, I was running into a situation where unrelated changes were causing cluster restarts. This logic couple with the "ForceNew" of install_script_action seems to do the right thing.